libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2657,6 +2992,15 @@ string YulUtilFunctions::storageSetToZeroFunction(Type const& _type)
 			("functionName", functionName)
 			("clearArray", clearStorageArrayFunction(dynamic_cast<ArrayType const&>(_type)))
 			.render();
+		else if (_type.category() == Type::Category::Struct)
```

Comment:
Ah, yes, you already have it here as well. This should now allow stuff like this to pass:
```
contract C {
  struct S { uint256 x; }
  S s;
  function f() public {
    s.x = 42;
    delete s;
    assert(s.x == 0);
  }
}
```
and (as mentioned in the other comment) can be its own PR with its own tests.

Reply:
#9843